compiler/rustc_target/src/spec/targets/i686_win7_windows_msvc.rs

```diff
@@ -1,18 +1,22 @@
-use crate::spec::{Arch, LinkerFlavor, Lld, RustcAbi, SanitizerSet, Target, TargetMetadata, base};
+use crate::spec::{
+    Arch, LinkerFlavor, Lld, RustcAbi, SanitizerSet, Target, TargetMetadata, TargetOptions, base,
+};
 
 pub(crate) fn target() -> Target {
-    let mut base = base::windows_msvc::opts();
-    base.vendor = "win7".into();
-    base.rustc_abi = Some(RustcAbi::X86Sse2);
-    base.cpu = "pentium4".into();
-    base.max_atomic_width = Some(64);
-    base.supported_sanitizers = SanitizerSet::ADDRESS;
-    // On Windows 7 32-bit, the alignment characteristic of the TLS Directory
-    // don't appear to be respected by the PE Loader, leading to crashes. As
-    // a result, let's disable has_thread_local to make sure TLS goes through
-    // the emulation layer.
-    // See https://github.com/rust-lang/rust/issues/138903
-    base.has_thread_local = false;
+    let mut base = TargetOptions {
+        vendor: "win7".into(),
+        rustc_abi: Some(RustcAbi::X86Sse2),
+        cpu: "pentium4".into(),
+        max_atomic_width: Some(64),
+        supported_sanitizers: SanitizerSet::ADDRESS,
+        // On Windows 7 32-bit, the alignment characteristic of the TLS Directory
+        // don't appear to be respected by the PE Loader, leading to crashes. As
+        // a result, let's disable has_thread_local to make sure TLS goes through
+        // the emulation layer.
+        // See https://github.com/rust-lang/rust/issues/138903
+        has_thread_local: false,
+        ..base::windows_msvc::opts()
+    };
```

Comment:
Nit: I'd prefer for these refactoring in the target definitions be done in a separate commit first.

(I assume you do it to not trigger the `rustc_lint_opt_deny_field_access` lint).